### PR TITLE
ON-40397 # add GOOGLE_MAPS integration

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

**Comment by mymattcarroll** [comment]
Are there form element changes to make as well? Or are we doing them in another ticket?

**Comment by divporter** [comment]
Josh has already pointed out that I missed it completely.  I just want the juicy apps work.  I will add it at some point.

File: typescript/forms.d.ts [hunk 1]
```diff
@@ -458,6 +458,15 @@ export type PointAddressElement = {
   FormElementRequired &
   FormElementReadOnly
 
+//TODO investigate filters
+export type GoogleAddressElement = {
+  type: 'googleAddress'
+  defaultValue?: string //TODO
```
Comment (mymattcarroll) [inline typescript/forms.d.ts:464]: Is there are another jira ticket for this or is it being done in this PR?

File: typescript/formsApps.d.ts [hunk 2]
```diff
@@ -501,6 +501,7 @@ export type FormsAppConfiguration<
   isGoogleLoginSupported: boolean
   isClientLoggingEnabled: boolean
   recaptchaPublicKey: string
+  googleMapsKey?: string
```
Comment (mymattcarroll) [inline typescript/formsApps.d.ts:504]: I know recaptcha is named like so, but can we update this to more accurately represent what is stored i.e.

```suggestion
  googleMapsIntegrationKeyId?: string
```

Comment (divporter) [reply typescript/formsApps.d.ts:504]: Ah yeah derp

File: typescript/formsApps.d.ts [hunk 3]
```diff
@@ -501,6 +501,7 @@ export type FormsAppConfiguration<
   isGoogleLoginSupported: boolean
   isClientLoggingEnabled: boolean
   recaptchaPublicKey: string
+  googleMapsIntegrationKeyId?: string
```
Comment (mymattcarroll) [inline typescript/formsApps.d.ts:504]: Whoops, this property should actually be what you had it as before. `googleMapsIntegrationKeyId` should be the property on the `FormsApp` type.

**Pull request merged**